src/momento-redis-adapter.ts

```diff
@@ -641,24 +643,72 @@ export class MomentoRedisAdapter
     if (args.length % 2 !== 0) {
       this.emitError(
         'mset',
-        "wrong number of arguments for 'mset' command",
+        "Wrong number of arguments for 'mset' command",
         MomentoErrorCode.INVALID_ARGUMENT_ERROR
       );
     }
+
+    const keys: RedisKey[] = [];
+    const maybeCompressedValues: (string | Promise<Uint8Array>)[] = [];
+
     for (let i = 0; i < args.length; i += 2) {
-      await this.set(args[i] as RedisKey, args[i + 1]);
+      keys.push(args[i] as RedisKey);
+      const value = args[i + 1];
+      if (this.useCompression) {
+        maybeCompressedValues.push(compress(value));
+      } else {
+        maybeCompressedValues.push(String(value));
+      }
+    }
+
+    let values: (string | Uint8Array)[];
+    if (this.useCompression) {
+      values = await Promise.all(maybeCompressedValues);
+    } else {
+      values = maybeCompressedValues as string[];
+    }
+
+    const items = new Map<RedisKey, Uint8Array | string>();
+    for (let i = 0; i < keys.length; i++) {
+      items.set(
+        keys[i],
+        this.useCompression ? values[i] : Buffer.from(values[i] as string)
+      );
+    }
+
+    const rsp = await this.momentoClient.setBatch(this.cacheName, items);
+    if (rsp instanceof CacheSetBatch.Success) {
+      return 'OK';
+    } else if (rsp instanceof CacheSetBatch.Error) {
+      this.emitError('mset', rsp.message(), rsp.errorCode());
+      return 'OK';
+    } else {
+      this.emitError('mset', `unexpected-response ${rsp.toString()}`);
+      return 'OK';
     }
-    return 'OK';
   }
 
   async mget(
     ...args: [key: RedisKey, ...keys: RedisKey[]]
   ): Promise<(string | null)[]> {
-    const promises: Promise<string | null>[] = [];
-    args.forEach(key => {
-      promises.push(this.get(key));
-    });
-    return await Promise.all(promises);
+    const resp = await this.momentoClient.getBatch(this.cacheName, args);
+
+    if (resp instanceof CacheGetBatch.Success) {
+      const keyValueRecord = resp.valuesRecordStringUint8Array();
+      const maybeCompressedValues = Object.values(keyValueRecord);
+      if (this.useCompression) {
+        return await Promise.all(
+          maybeCompressedValues.map(async value => {
+            return await decompress(value);
+          })
+        );
+      } else {
+        return Object.values(resp.values());
+      }
+    } else if (resp instanceof CacheGetBatch.Error) {
+      this.emitError('mget', resp.message(), resp.errorCode());
+    }
+    return [];
```

Comment:
what is the case where we would get here? a Miss? If so, It's probably better to have an explicit branch for that, and then a final explicit `else` branch that throws an error if `resp` doesn't match one of the known types.